Item12849: Add Wildcard ACL support in support of deprecating blank DENY ACLs

This patch enables ALLOW and DENY ACLs to use the "*" wildcard to match any user

~~~PATCH 3a40329f9e882fe958ec77991d845d5f:b776aa68d6fd19a166718dacd6f3e762 lib/Foswiki/Users.pm ( Foswiki 1.1.1-1.1.3 )
--- lib/Foswiki/Users.pm	2010-10-25 17:58:13.000000000 -0400
+++ lib/Foswiki/Users.pm	2014-04-11 20:07:01.767296086 -0400
@@ -636,6 +636,11 @@
 
     foreach my $ident (@$userlist) {
 
+        # The Wildcard match.  Any user matches the "*" identifier.
+        if ( $ident eq '*' ) {
+            return 1;
+        }
+
         my $identCUID = $this->getCanonicalUserID($ident);
 
         if ( defined $identCUID ) {

